### ISSUE TITLE
False positive indentation result when the semi-colon is at the beginning of a new line.

### ISSUE THREAD

**Issue by mortargrind** [eslint/eslint#13833]
**Tell us about your environment**

* **ESLint Version: 7.13.0**
* **Node Version: 14.8.0**
* **npm Version: 6.14.8**

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
esprima (default)

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parser: 'esprima',

  parserOptions: {
    ecmaVersion: 2019,
    sourceType: 'module',
    ecmaFeatures: {
      jsx: true,
    },
  },

  env: {
    browser: true,
  },

  settings: {},

  rules: {
    'semi': ['error', 'never', { 'beforeStatementContinuationChars': 'always' }],
    'indent': ['error', 2],
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function randomFunction() {
  return {
    aa: 4,
    bb: 5,
    cc: 6
  }
}

function main() {
  let aa, bb, cc

  const result = randomFunction()

         ;({
    aa,
    bb,
    cc,
  } = result)
}

export {
  main,
}

```

```bash
./node_modules/.bin/eslint ./ --ext .js
```

**What did you expect to happen?**
It should give an error saying `Expected intendtion of 2 spaced but found 9` 

**What actually happened? Please include the actual, raw output from ESLint.**
It does not report any error.

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by anikethsaha** [comment]
thanks for the issue. 

Looks like a bug to me.  
Not sure whether it is intentional or not. If there is a semi-colon first line, it seems not indenting for some case as per [here](https://github.com/eslint/eslint/blob/master/lib/rules/indent.js#L907)

[Demo](https://eslint.org/demo#eyJ0ZXh0IjoiZnVuY3Rpb24gbWFpbigpIHtcbiAgbGV0IGFhLCBiYiwgY2NcblxuICBjb25zdCByZXN1bHQgPSByYW5kb21GdW5jdGlvbigpXG4gXG4gICAgICAgICA7Zm4oKVxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTIsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiaW5kZW50IjoyLCJmdW5jdGlvbi1wYXJlbi1uZXdsaW5lIjoyfSwiZW52Ijp7fX19)
[Demo 2](https://eslint.org/demo#eyJ0ZXh0IjoiZnVuY3Rpb24gbWFpbigpIHtcbiAgbGV0IGFhLCBiYiwgY2NcblxuICBjb25zdCByZXN1bHQgPSByYW5kb21GdW5jdGlvbigpXG5mbigpXG4gICAgICAgICA7Zm4oKVxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTIsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiaW5kZW50IjoyLCJmdW5jdGlvbi1wYXJlbi1uZXdsaW5lIjoyfSwiZW52Ijp7fX19)
